### PR TITLE
feat: new config format

### PR DESCRIPTION
**Pull request by RomanHotsiy**
New universal config support.

@knidarkness review please
@adamaltman review please deprecation messages language

### CODE REVIEW

File: src/cli/index.js [hunk 1]
```diff
@@ -89,37 +95,41 @@ const cli = () => {
...
+      if (entryPoints.length > 1) {
         process.stdout.write(`Total results. Errors: ${results.errors}, warnings: ${results.warnings}\n`);
       }
       process.exit(results.errors > 0 ? 1 : 0);
     });
 
   program.on('command:*', () => {
-  	process.stdout.write(`\nUnknown command ${program.args.join(' ')}\n\n`);
-  	program.outputHelp();
+    process.stdout.write(`\nUnknown command ${program.args.join(' ')}\n\n`);
```
Comment (adamaltman) [inline src/cli/index.js]: Normal output or an error? (stdout vs. stderr)

File: src/config.js [hunk 2]
```diff
@@ -1,53 +1,95 @@
...
-    }
-
-    if (fs.existsSync(`${process.cwd()}/.openapi-cli.yml`)) {
-      configPath = `${process.cwd()}/.openapi-cli.yml`;
+    if (fs.existsSync('.redocly.yaml')) {
+      configPath = path.resolve('.redocly.yaml');
+    } else if (fs.existsSync('.redocly.yml')) {
+      configPath = path.resolve('.redocly.yml');
+    } else if (fs.existsSync('.openapi-cli.yaml')) {
+      if (!warningShown) process.stderr.write('warning: .openapi-cli.yaml is deprecated, rename to .redocly.yaml\n');
```
Comment (adamaltman) [inline src/config.js]: This misses that the actual structure of the file has changed too -- it should have `lint:`

Comment (adamaltman) [reply src/config.js]: Nevermind, I see you also have that message below.

File: src/config.js [hunk 3]
```diff
@@ -1,53 +1,95 @@
...
 
   if (fs.existsSync(configPath)) {
     const configRaw = fs.readFileSync(configPath, 'utf-8');
     config = yaml.safeLoad(configRaw);
+
+    if (config.rules || config.transformers || config.typeExtension || config.customRules) {
+      if (!warningShown) {
+        process.stderr.write(
+          'warning: top level "rules", "transformers", "typeExtension" and "customRules" '
+          + 'are deprecated. Move them under "lint" field\n',
```
Comment (adamaltman) [inline src/config.js]:
```suggestion
          + 'are deprecated. Move them under the "lint" field\n',
```

File: src/config.js [hunk 4]
```diff
@@ -1,53 +1,95 @@
...
+  if (
+    (!argsEntrypoints || !argsEntrypoints.length)
+    && config.apiDefinitions
+    && Object.keys(config.apiDefinitions).length > 0
+  ) {
+    res = Object.values(config.apiDefinitions);
+  }
+
+  if (!res || !res.length) {
+    process.stdout.write('error: missing required argument "entryPoints"\n');
```
Comment (adamaltman) [inline src/config.js]: Same question throughout -- stdout vs. stderr?

**Comment by RomanHotsiy** [comment]
Opened https://github.com/Redocly/openapi-cli/issues/88

**Pull request merged**